health_check/contrib/celery/backends.py

```diff
@@ -1,29 +1,69 @@
+from celery.app import default_app as app
 from django.conf import settings
 
 from health_check.backends import BaseHealthCheckBackend
-from health_check.exceptions import (
-    ServiceReturnedUnexpectedResult, ServiceUnavailable
-)
-
-from .tasks import add
+from health_check.exceptions import ServiceUnavailable
 
 
 class CeleryHealthCheck(BaseHealthCheckBackend):
-    def check_status(self):
-        timeout = getattr(settings, 'HEALTHCHECK_CELERY_TIMEOUT', 3)
+    CORRECT_RESPONSE = {"ok": "pong"}
+
+    def _check_active_queues(self):
+        defined_queues = app.conf.CELERY_QUEUES
+
+        if not defined_queues:
+            return
+
+        active_queues = []
+
+        for queues in list(
+            app.control.inspect(self.active_workers).active_queues().values()
+        ):
+            active_queues.extend([queue.get("name") for queue in queues])
+
+        for defined_queue in defined_queues:
+            if defined_queue.name not in active_queues:
+                self.add_error(
+                    ServiceUnavailable(
+                        f"No worker for Celery task queue {defined_queue.name}"
+                    ),
+                )
 
+    def _check_ping_result(self):
+        self.active_workers = []
```

Comment:
To avoid setting instance attributes outside of `__init__` method, let's modify `_check_acvite_queues` so it will accept `actove_workers` argument